models/glif_psc.h

```diff
@@ -0,0 +1,354 @@
+/*
+ *  glif_psc.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef GLIF_PSC_H
+#define GLIF_PSC_H
+
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+#include "dictdatum.h"
+
+/* BeginDocumentation
+Name: glif_psc - Current-based generalized leaky integrate and fire (GLIF)
+models
+
+Description:
+
+glif_psc is implementations of five generalized leaky integrate and
+fire (GLIF) models [1] with alpha-function shaped synaptic currents.
+Incoming spike events induce a post-synaptic change of current modeled
+by an alpha function. The alpha function is normalized such that an event of
+weight 1.0 results in a peak current of 1 pA at t = tau_syn. On the postsynapic
+side, there can be arbitrarily many synaptic time constants. This can be reached
+by specifying separate receptor ports, each for a different time constant.
+The port number has to match the respective "receptor_type" in the connectors.
+
+The five GLIF models are
+
+GLIF model 1 - Traditional leaky integrate and fire (LIF) model,
+GLIF model 2 - Leaky integrate and fire with biologically defined reset rules
+               model,
+GLIF model 3 - Leaky integrate and fire with after-spike currents model,
+GLIF model 4 - Leaky integrate and fire with biologically defined reset rules
+               and after-spike currents model,
+GLIF model 5 - Leaky integrate and fire with biologically defined reset rules,
+               after-spike currents and a voltage dependent threshold model.
+
+Parameters:
+
+The following parameters can be set in the status dictionary.
+
+V_m        double - Membrane potential in mV
+V_th       double - Instantaneous threshold in mV.
+g          double - Membrane conductance in nS.
+E_L        double - Resting membrane potential in mV.
+C_m        double - Capacitance of the membrane in pF.
+t_ref      double - Duration of refractory time in ms.
+V_reset    double - Reset potential of the membrane in mV (glif 1-3).
+a_spike    double - Threshold addition following spike in mV.
```

Comment:
Which parameter is this in Table 1 in [1]? Should be explicit for all parameters.